### PR TITLE
refactor(prune-types/prune): move PruneLimiter to `reth-prune`

### LINKED ISSUE #13149: Move PruneLimiter to prune crate
### Describe the feature

This type currentl resides in the prune/types crate but we only need this in prune crate:

https://github.com/paradigmxyz/reth/blob/08c9df8cb37300b45fdc107fe37163f04429e833/crates/prune/types/src/limiter.rs#L9-L9

we should move this because we need this for #13041

and we also need to flip this function:

https://github.com/paradigmxyz/reth/blob/08c9df8cb37300b45fdc107fe37163f04429e833/crates/prune/types/src/pruner.rs#L105-L105

and impl this for Prunelimiter instead

same here:

https://github.com/paradigmxyz/reth/blob/08c9df8cb37300b45fdc107fe37163f04429e833/crates/prune/types/src/pruner.rs#L132-L132

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lean-apple**
Closes https://github.com/paradigmxyz/reth/issues/13149. 

I tried to limit the changes @mattsse.

### CODE REVIEW

File: crates/prune/prune/src/segments/user/storage_history.rs [hunk 1]
```diff
@@ -65,7 +62,7 @@ where
         };
         if limiter.is_limit_reached() {
             return Ok(SegmentOutput::not_done(
-                PruneInterruptReason::new(&limiter),
+                PruneLimiter::interrupt_reason(&limiter),
```
Comment (mattsse) [inline crates/prune/prune/src/segments/user/storage_history.rs:65]: let's replace these with `limiter.interrupt_reason()`

File: crates/prune/prune/src/segments/user/storage_history.rs [hunk 2]
```diff
@@ -118,7 +115,7 @@ where
         )?;
         trace!(target: "pruner", ?outcomes, %done, "Pruned storage history (indices)");
 
-        let progress = PruneProgress::new(done, &limiter);
+        let progress = PruneLimiter::progress(&limiter, done);
```
Comment (mattsse) [inline crates/prune/prune/src/segments/user/storage_history.rs:118]: same

**Review by mattsse — CHANGES_REQUESTED** [review]
ty!

smol style nit re function call notation

File: crates/prune/prune/src/segments/user/storage_history.rs (hunk 1 above)
Comment (lean-apple) [reply crates/prune/prune/src/segments/user/storage_history.rs]: Ah yes, it's much better.

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
ty!

**Pull request merged**